nixos/modules/services/x11/desktop-managers/pantheon.nix

```diff
@@ -168,12 +168,35 @@ in
       services.udev.packages = [
         pkgs.gnome.gnome-settings-daemon338
       ];
-      systemd.packages = [
-        pkgs.gnome.gnome-settings-daemon338
+      systemd.packages = with pkgs; [
+        gnome.gnome-settings-daemon338
+        gnome.gnome-session
+        pantheon.gala
+        pantheon.elementary-session-settings
       ];
       programs.dconf.enable = true;
       networking.networkmanager.enable = mkDefault true;
 
+      systemd.user.targets."gnome-session-initialized".wants = [
+        "org.gnome.SettingsDaemon.A11ySettings.target"
+        "org.gnome.SettingsDaemon.Color.target"
+        "org.gnome.SettingsDaemon.Datetime.target"
+        "org.gnome.SettingsDaemon.Housekeeping.target"
+        "org.gnome.SettingsDaemon.Keyboard.target"
+        "org.gnome.SettingsDaemon.MediaKeys.target"
+        "org.gnome.SettingsDaemon.Power.target"
+        "org.gnome.SettingsDaemon.PrintNotifications.target"
+        "org.gnome.SettingsDaemon.Rfkill.target"
+        "org.gnome.SettingsDaemon.ScreensaverProxy.target"
+        "org.gnome.SettingsDaemon.Sharing.target"
+        "org.gnome.SettingsDaemon.Smartcard.target"
+        "org.gnome.SettingsDaemon.Sound.target"
+        "org.gnome.SettingsDaemon.UsbProtection.target"
+        "org.gnome.SettingsDaemon.Wacom.target"
+        "org.gnome.SettingsDaemon.Wwan.target"
+        "org.gnome.SettingsDaemon.XSettings.target"
+      ];
+
```

Comment:
Maybe this can fix the media key issue (cc @davidak) but I am not sure if this is the correct place to put this... Shall I update [`nixos/modules/services/desktops/gnome/gnome-settings-daemon.nix`](https://github.com/NixOS/nixpkgs/blob/bb36c044d1be1be5f023fc9f4a5dc38b5c023787/nixos/modules/services/desktops/gnome/gnome-settings-daemon.nix#L60-L80) instead (cc @jtojnar) or shall this included in upstream (cc @davidmhewitt)?

Reply:
It's been a while since I've looked at this, and I don't think I fully understand what you're doing with the nix packaging here, but the way this is supposed to work is that the `gnome-settings-daemon` package installs a set of systemd units and targets that automatically start with `gnome-session` targets.

Since we launch Pantheon with `gnome-session`, all of the GNOME SettingsDaemon plugins should just work without us explicitly having to specify we want them to start, so I don't think there's anything to do "upstream" in elementary here. But please let us know what you find.

Is this potentially an issue where different versions of GNOME components are being used together for some reason?

Reply:
@davidmhewitt Haven't tested this, but I found this in `/etc/systemd/user/gnome-session@gnome.target.d/gnome.session.conf`:

```
[Unit]
# Must be in sync with gnome.session
Wants=org.gnome.SettingsDaemon.A11ySettings.target
Wants=org.gnome.SettingsDaemon.Color.target
Wants=org.gnome.SettingsDaemon.Datetime.target
Wants=org.gnome.SettingsDaemon.Housekeeping.target
Wants=org.gnome.SettingsDaemon.Keyboard.target
Wants=org.gnome.SettingsDaemon.MediaKeys.target
Wants=org.gnome.SettingsDaemon.Power.target
Wants=org.gnome.SettingsDaemon.PrintNotifications.target
Wants=org.gnome.SettingsDaemon.Rfkill.target
Wants=org.gnome.SettingsDaemon.ScreensaverProxy.target
Wants=org.gnome.SettingsDaemon.Sharing.target
Wants=org.gnome.SettingsDaemon.Smartcard.target
Wants=org.gnome.SettingsDaemon.Sound.target
Wants=org.gnome.SettingsDaemon.UsbProtection.target
Wants=org.gnome.SettingsDaemon.Wacom.target
Wants=org.gnome.SettingsDaemon.XSettings.target
```

I thought there should have something similar for `gnome-session@pantheon.target`...

Reply:
Filed https://github.com/elementary/session-settings/pull/63